Mathlib/Data/Matrix/ColumnRowPartitioned.lean

```diff
@@ -0,0 +1,229 @@
+  funext i j
+  unfold fromRows
+  simp only [Sum.elim_zero_zero, of_apply, Pi.zero_apply, zero_apply]
+
+lemma fromColumns_zero : fromColumns (0 : Matrix M N₁ R) (0 : Matrix M N₂ R) = 0 := by
+  funext i j
+  unfold fromColumns
+  cases' j with j j
+  all_goals (simp only [of_apply, Sum.elim_inl, Sum.elim_inr, zero_apply])
+
+/-- A row partitioned matrix multiplied by a column partioned matrix gives a 2 by 2 block matrix -/
+lemma fromRows_mul_fromColumns (A₁ : Matrix M₁ N R) (A₂ : Matrix M₂ N R)
+    (B₁ : Matrix N N₁ R) (B₂ : Matrix N N₂ R) :
+    (fromRows A₁ A₂) ⬝ (fromColumns B₁ B₂) = fromBlocks (A₁⬝B₁) (A₁⬝B₂) (A₂⬝B₁) (A₂⬝B₂) := by
+  funext i j
+  rw [fromRows, fromColumns]
+  cases i;
+  all_goals (cases j)
+  all_goals (simp only [
+    fromBlocks_apply₁₁, fromBlocks_apply₁₂, fromBlocks_apply₂₁, fromBlocks_apply₂₂,
+      mul_apply, of_apply, Sum.elim_inl, Sum.elim_inr] )
+
+/-- A column partitioned matrix mulitplied by a row partitioned matrix gives the sum of the "outer"
+products of the block matrices -/
+lemma fromColumns_mul_fromRows (A₁ : Matrix M N₁ R) (A₂ : Matrix M N₂ R)
+    (B₁ : Matrix N₁ N R) (B₂ : Matrix N₂ N R) :
+    fromColumns A₁ A₂ ⬝ fromRows B₁ B₂ = (A₁⬝B₁ + A₂⬝B₂) := by
+  funext i j
+  rw [fromRows, fromColumns]
+  simp only [add_apply, mul_apply, of_apply, Fintype.sum_sum_type, Sum.elim_inl, Sum.elim_inr]
+
+/-- A column partitioned matrix multipiled by a block matrix results in a column partioned matrix -/
+lemma fromColumns_mul_fromBlocks (A₁ : Matrix M M₁ R) (A₂ : Matrix M M₂ R)
+    (B₁₁ : Matrix M₁ N₁ R) (B₁₂ : Matrix M₁ N₂ R) (B₂₁ : Matrix M₂ N₁ R) (B₂₂ : Matrix M₂ N₂ R) :
+    (fromColumns A₁ A₂) ⬝ fromBlocks B₁₁ B₁₂ B₂₁ B₂₂ =
+      fromColumns (A₁⬝B₁₁ + A₂⬝B₂₁) (A₁⬝B₁₂ + A₂⬝B₂₂) := by
+  funext i j
+  rw [fromColumns, fromColumns, fromBlocks]
+  cases j
+  all_goals simp only [of_apply, add_apply, Fintype.sum_sum_type, Sum.elim_inl,
+    Sum.elim_inr, mul_apply]
+
+/-- A block matrix mulitplied by a row partitioned matrix gives a row partitioned matrix -/
+lemma fromBlocks_mul_fromRows (A₁ : Matrix N₁ N R) (A₂ : Matrix N₂ N R)
+    (B₁₁ : Matrix M₁ N₁ R) (B₁₂ : Matrix M₁ N₂ R) (B₂₁ : Matrix M₂ N₁ R) (B₂₂ : Matrix M₂ N₂ R) :
+    fromBlocks B₁₁ B₁₂ B₂₁ B₂₂ ⬝ (fromRows A₁ A₂) =
+      fromRows (B₁₁⬝A₁ + B₁₂⬝A₂) (B₂₁⬝A₁ + B₂₂⬝A₂) := by
+  funext i j
+  rw [fromRows, fromRows, fromBlocks]
+  cases i
+  all_goals simp only [of_apply, add_apply, Fintype.sum_sum_type, Sum.elim_inl, Sum.elim_inr,
+    mul_apply]
+
+end Semiring
+
+section CommRing
+
+variable [CommRing R]
+
+/- Given that the index set N and the direct sum of the index sets N₁ and N₂ are in bijection then
+the matrix A : N × (N₁ ⊕ N₂) := Cols[A₁ A₂] is actually a "square". Hence, if its product with
+another matrix B : (N₁ ⊕ N₂) × N := Rows[B₁ B₂] matrix is one, the other matrix must be its inverse
+Mulitplication of a matrix by its inverse is commutative. This is column(row) partioned matrix form
+of `mul_eq_one_comm` -/
```

Comment:
I didn't really understand the comment you wrote. There seem to be four main messages in here:
 * A high-level description of the theorem
 * How it relates to other theorems
 * What the bijection means
 * What the theorem precisely states.

All of these are valuable information, but I would present them in this order. Moreover, what the theorem precisely states can be read from the type of the theorem so there is no need to repeat that in the comment.

```suggestion
/-- Multiplication of a matrix by its inverse is commutative.
This is the column and row partitioned matrix form of `Matrix.mul_eq_one_comm`.

The condition `e : N ≃ N₁ ⊕ N₂` states that `fromColumns A₁ A₂` and `fromRows B₁ B₂` are "square".
-/
```